pkg/sentry/socket/netlink/route/protocol_unsafe.go

```diff
@@ -0,0 +1,31 @@
+// Copyright 2019 The gVisor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package route provides a NETLINK_ROUTE socket protocol.
+package route
+
+import (
+	"syscall"
+	"unsafe"
+
+	"gvisor.dev/gvisor/pkg/syserr"
+)
+
+func parseDstAddr(b []byte) ([]byte, *syserr.Error) {
+        a := (*syscall.RtAttr)(unsafe.Pointer(&b[0]))
```

Comment:
Where is this type? It is not in the Go standard library: https://github.com/golang/go/search?q=RtAddr&unscoped_q=RtAddr

Also, why does this need to use an unsafe cast? Please use a safe unmarshal if possible.

Reply:
https://golang.org/pkg/syscall/#RtAttr
https://github.com/golang/go/blob/cedb561617ea40c72dc018b38c4d5f0862595fe8/src/syscall/types_linux.go#L361

The comment about unsafe cast stands though.

Reply:
First of all, sorry for delayed response.

Will supersede unsafe cast with unmarshal. And put those definitions in abi/linux.